### PR TITLE
save excluded groups in json format 

### LINKED ISSUE #10983: Exclude with groups cannot work with groups containing commas
### Steps to reproduce:
1. Create a group called "comma,group"
2. Exclude that group from sharing

### Expected result
Group properly excluded

### Actual result
(not tested but guessing) group name is split into two groups "comma" and "group", so exclusion cannot work.

Note that other places like the "enable app for groups" use JSON to store the groups array.

The "exclude sharing for groups" feature need to be fixed to also use JSON.

CC @schiesbn

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #10983

@PVince81

### CODE REVIEW

**Comment by karlitschek** [comment]
great :+1:

File: lib/private/util.php [hunk 1]
```diff
@@ -223,7 +223,12 @@ public static function isSharingDisabledForUser() {
 		if (\OC::$server->getAppConfig()->getValue('core', 'shareapi_exclude_groups', 'no') === 'yes') {
 			$user = \OCP\User::getUser();
 			$groupsList = \OC::$server->getAppConfig()->getValue('core', 'shareapi_exclude_groups_list', '');
-			$excludedGroups = explode(',', $groupsList);
+			$excludedGroups = json_decode($groupsList);
+			if (is_null($excludedGroups)) {
+				$excludedGroups = explode(',', $groupsList);
+				$newValue = json_encode($excludedGroups);
+				\OC::$server->getAppConfig()->setValue('core', 'shareapi_exclude_groups_list', $newValue);
```
Comment (DeepDiver1975) [inline lib/private/util.php]: @PVince81 this is the migration step :see_no_evil:

**Comment by PVince81** [comment]
Works :+1:

File: lib/private/util.php (hunk 1 above)
Comment (MorrisJobke) [inline lib/private/util.php]: Why then not add it as an repair step? :see_no_evil:

Comment (PVince81) [inline lib/private/util.php]: On-demand fixing is the new cool thing

Comment (MorrisJobke) [inline lib/private/util.php]: And leaves a lot of code just because it could be not fixed in a specific use case :(

**Comment by LukasReschke** [comment]
Regression: https://github.com/owncloud/core/issues/19542

**Pull request merged**